### PR TITLE
OAM-200: Delele orders

### PR DESCRIPTION
**Pull request by tsznaj**
(cherry picked from commit fd63632492b7dbf315bcf6be89c5b52f822b5306)

### CODE REVIEW

File: src/main/java/org/openlmis/fulfillment/web/OrderController.java [hunk 1]
```diff
@@ -503,6 +505,26 @@ public ResultDto<Boolean> retryOrderTransfer(@PathVariable("id") UUID id) {
     return new ResultDto<>(TRANSFER_FAILED != order.getStatus());
   }
 
+  @RequestMapping(value = "/orders", method = RequestMethod.DELETE)
+  @ResponseStatus(HttpStatus.OK)
+  public void deleteMultipleOrders(@RequestParam(name = "ids") List<UUID> ids) {
+
+    if (CollectionUtils.isEmpty(ids)) {
+      XLOGGER.info("Nothing to delete");
```
Comment (pwargulak) [inline src/main/java/org/openlmis/fulfillment/web/OrderController.java:513]: Missing an early exit?

**Review by pwargulak — CHANGES_REQUESTED** [review]
I'm missing CREATING-status check for Order deletion.

@sradziszewski do permissions have migration scripts?

**Comment by sradziszewski** [comment]
> I'm missing CREATING-status check for Order deletion.
> 
> @sradziszewski do permissions have migration scripts?

Yes, if we want to check it this way, we need to add the ORDER_DELETE right using migration.

**Review by sradziszewski — COMMENTED** [review]
Add migration with new right insertion.

**Comment by tsznaj** [comment]
Added rights here: https://github.com/OpenLMIS/openlmis-referencedata/pull/105

**Review by pwargulak — CHANGES_REQUESTED** [review]
Still can't see how the 3rd AC is solved:

> Orders must have the status CREATING  in order to be eligible for deletion. Other statuses will not be accepted.

File: src/integration-test/java/org/openlmis/fulfillment/web/OrderControllerIntegrationTest.java [hunk 2]
```diff
@@ -499,6 +498,44 @@ public void shouldCreateOrder() {
...
+    firstOrder.setStatus(OrderStatus.CREATING);
+    secondOrder.setStatus(OrderStatus.CREATING);
+
+    UUID firstOrderId = firstOrder.getId();
+    UUID secondOrderId = secondOrder.getId();
+
+    List<UUID> uuids = new ArrayList<>();
+    uuids.add(firstOrderId);
+    uuids.add(secondOrderId);
+
```
Comment (sradziszewski) [inline src/integration-test/java/org/openlmis/fulfillment/web/OrderControllerIntegrationTest.java:513]: Remove redundant empty line.

**Review by sradziszewski — APPROVED** [review]

File: src/integration-test/java/org/openlmis/fulfillment/web/OrderControllerIntegrationTest.java (hunk 2 above)
Comment (tsznaj) [reply src/integration-test/java/org/openlmis/fulfillment/web/OrderControllerIntegrationTest.java:513]: funny comment after so many crazy commits :D

**Pull request merged**